packages/PHPStanStaticTypeMapper/TypeMapper/IntersectionTypeMapper.php

```diff
@@ -52,19 +51,7 @@ public function getNodeClass(): string
      */
     public function mapToPHPStanPhpDocTypeNode(Type $type): TypeNode
     {
-        $intersectionTypesNodes = [];
-
-        foreach ($type->getTypes() as $intersectionedType) {
-            $intersectionTypesNodes[] = $this->phpStanStaticTypeMapper->mapToPHPStanPhpDocTypeNode($intersectionedType);
-        }
-
-        $intersectionTypesNodes = array_unique($intersectionTypesNodes);
-
-        if (count($intersectionTypesNodes) === 1) {
-            return $intersectionTypesNodes[0];
-        }
-
-        return new BracketsAwareIntersectionTypeNode($intersectionTypesNodes);
+        return $type->toPhpDocNode();
```

Comment:
@TomasVotruba this cause issue : 

- https://github.com/rectorphp/rector/issues/8086

which fqcn got invalid non-fqcn classes.